modules/restful_token_auth/plugins/authentication/RestfulAuthenticationToken.class.php

```diff
@@ -12,9 +12,11 @@ class RestfulAuthenticationToken extends \RestfulAuthenticationBase {
   public function applies(array $request = array(), $method = \RestfulInterface::GET) {
     $options = $this->getPluginKey('options');
     $key_name = !empty($options['param_name']) ? $options['param_name'] : 'access_token';
+    $dashed_key_name = str_replace('_', '-', $key_name);
 
-    // Access token may be on the request, or in the headers.
-    return !empty($request['__application'][$key_name]) || !empty($request[$key_name]);
+    // Access token may be on the request, or in the headers
+    // (may be a with dash instead of underscore).
+    return !empty($request['__application'][$key_name]) || !empty($request[$key_name]) || !empty($request['__application'][$dashed_key_name]) || !empty($request[$dashed_key_name]);
```

Comment:
Maybe better adapt `restful_extract_token_from_request()` to return FALSE if nothing found, to avoid this `!empty()` checks.